migrate-abs-to-spe.md

```diff
@@ -0,0 +1,207 @@
+
+# Authenication
+
+## Azure Blob Storage
+
+1. Credentials - Container level Shared Access Signature (SAS) URL.
+1. Permission - Read and List
+
+## SharePoint Embedded site
+
+1. App type - A guest app
+1. Credentials - SPE client id and SPE container
+    -  Have permissions to Write and Create folder in the container
+1. Permission - "User.Read", "FileStorage.Container.Selected", "Files.ReadWrite"
+
+# Overview of the Sample App
+
+## Description
+
+The sample app is called "MigrateABStoSPE" and it is designed to migrate files from an Azure Blob Storage (ABS) container to a SharePoint Online (SPE) container. It uses Azure.Storage.Blobs and Newtonsoft.Json libraries for working with ABS and JSON data respectively. The app authenticates with both ABS and SPE using client credentials and performs the migration of files.
+
+## Running the Sample App
+
+1.	Open a terminal or command prompt.
+1.	Navigate to the directory where the Program.cs file is located.
+1.	Make sure you have the .NET Core SDK installed on your machine. You can check this by running the command dotnet --version in the terminal. If the command is not recognized, you can download and install the .NET Core SDK from the official Microsoft website.
+1.	Once you have confirmed that the .NET Core SDK is installed, you can build the application by running the command `dotnet build`. This will compile the code and generate the necessary binaries.
+1.	After the build process is complete, you can run the application by executing the command dotnet run followed by the required arguments. The required arguments are:
+    - The container-level SAS URL: This is an Azure Blob container level SAS URL. It provides access to the container and its blobs.
+    - The SPE client ID: This is the client ID that you are authenticating against in the SPE.
+    - The SPE container ID: This is the container ID that you are migrating content to in the SPE.
+    - (Optional) The list of blobs you want to copy in JSON format: This is an optional argument that allows you to specify a list of blobs to copy. The format should be an array of strings in JSON format.
+
+For example, the command to run the application with the required arguments would look like this:
+
+   `dotnet run Program.cs -s "<sas url>" -c "<clientid>" -o "<container id>" [-b "[\""example1.txt\"", \""example2.txt\"", \""example3.txt\""]"]`
+
+## Code Structure
+
+The code is organized into a single file named "Program.cs" within the "MigrateABStoSPE" namespace. The file contains the main entry point of the application, which is the Main method. It also includes several helper methods and a class definition for the Options class.
+
+The Options class is a nested class within the Program class and defines the command-line options that the app accepts. These options include the container-level SAS URL, the SPE client ID, the SPE container ID, and an optional list of blobs to copy in JSON format.
```

Comment:
as a user trying to learn how to copy data from ABS to SPE, I don’t care about this